diff --git a/program/gbuffers_entities.glsl b/program/gbuffers_entities.glsl
--- a/program/gbuffers_entities.glsl
+++ b/program/gbuffers_entities.glsl
@@ -18,4 +18,8 @@
 #ifdef FRAGMENT_SHADER
 
 in vec2 texCoord;
+#ifdef CRANKSHAFT_CLRWL_ENTITIES
+vec2 lmCoord;
+#else
 in vec2 lmCoord;
+#endif
@@ -162,9 +166,21 @@
     if (glColor.a < 0.0) discard;
+#ifndef CRANKSHAFT_CLRWL_ENTITIES
     skyLightCheck = pow2(1.0 - min1(lmCoord.y * 2.9 * sunVisibility));
+#endif
     #if SHOCKWAVE > 0
         vec4 color = doShockwave(playerPos + relativeEyePosition, texCoord);
     #else
         vec4 color = texture2D(tex, texCoord);
     #endif
     vec3 colorP = color.rgb;
+#ifdef CRANKSHAFT_CLRWL_ENTITIES
+    float clrwlAo;
+    vec4 clrwlOverlay;
+    clrwl_computeFragment(color, color, lmCoord, clrwlAo, clrwlOverlay);
+    color.rgb = mix(color.rgb, clrwlOverlay.rgb, clrwlOverlay.a);
+    lmCoord = clamp((lmCoord - 1.0 / 32.0) * 32.0 / 30.0, 0.0, 1.0);
+    lmCoord.x = min(lmCoord.x, 0.9);
+    skyLightCheck = pow2(1.0 - min1(lmCoord.y * 2.9 * sunVisibility));
+#else
     color *= glColor;
+#endif
